src/mlib/path.h

```diff
@@ -7,66 +7,151 @@
 
 #include <inttypes.h>
 
-/**
- * @brief The preferred path separator for the current platform
- */
-static const char MPATH_PREFERRED_PATH_SEPARATOR =
+#ifndef _WIN32
+#include <unistd.h>
+#else
+#include <windows.h>
+#endif
+
+typedef enum mpath_format {
```

Comment:
Curious, was the motivation for making the path format a runtime choice to ease development? I.e. you no longer need to use a Windows machine to test this?